File: py/Level0/다항식 더하기.py
#
# [level 0] 다항식 더하기
# https://school.programmers.co.kr/learn/courses/30/lessons/120863

# 결과
# 정확성: 100.0
# 합계: 100.0 / 100.0

def solution(polynomial):
    num = polynomial.split(' + ')
    x, n = 0, 0

    for i in num:
        if 'x' in i:
            if len(i) > 1:
                x += int(i[:-1])
            else:
                x += 1
        else:
            n += int(i)

    if x == 0:
        return str(n)
    elif x == 1:
        if n == 0:
            return 'x'
        else:
            return "x + {0}".format(n)
    elif x > 1:
        if n == 0:
            return "{0}x".format(x)
        else:
            return "{0}x + {1}".format(x, n)

File: js/lv0/다항식 더하기.js
//질문

//[level 0] 다항식 더하기 120863
//https://school.programmers.co.kr/learn/courses/30/lessons/120863

//결과 
//정확성: 58.3
//합계: 58.3 / 100.0

function solution(polynomial) {
    var num = polynomial.split(' + ');
    var x = 0;
    var n = 0;
    
    for (let i of num) {
        if (i.endsWith('x')) {
            if (i.length > 1) {
                x += Number(i.charAt(0));
            } else x += 1;
        } else n += Number(i);
    }
    
    //출력
    var result = [];
    if (x) result.push(`${x === 1? "" : x}x`);
    if (n) result.push(n);
        
    return result.join(" + ");
}
